### PR TITLE
re-gen from the latest spec

### PR DESCRIPTION
**Pull request by devoncarew**
- re-gen from the latest spec
- remove some special casing no longer required (after spec updates)

### CODE REVIEW

**Comment by devoncarew** [comment]
Updated to address some lint warnings, and make the CI system fail on lints.

**Comment by devoncarew** [comment]
Updated to handle generating empty constructors.

**Comment by devoncarew** [comment]
Add two new `fixedId` locations to the spec.

**Comment by devoncarew** [comment]
@jakemac53, this PR has a few more concerns in it than I'd originally intended, however:

- commit the new spec updates from Ben
- re-gen
- remove some special casing no longer required (after spec updates)
- address some lint issues found on the bot
- make the bot fail on info level lint issues
- update the spec to call out two additional places where we need to add `fixedId`s

File: dart/tool/service.md [hunk 1]
```diff
@@ -2071,6 +2076,11 @@ class Isolate extends Response {
   // isolate.
   string id;
 
+  // Provided and set to true if the id of an Object is fixed. If true, the id
+  // of an Object is guaranteed not to change or expire. The object may, however,
+  // still be _Collected_.
+  bool fixedId [optional];
```
Comment (devoncarew) [inline dart/tool/service.md]: @bkonyi, here (and for IsolateRef), I think we need to add explicit `fixedId`s to the spec.

File: dart/tool/dart/generate_dart.dart [hunk 2]
```diff
@@ -1069,26 +1069,6 @@ class Type extends Member {
 
   Type(this.parent, String categoryName, String definition, [this.docs]) {
     _parse(new Tokenizer(definition).tokenize());
-    // Anything with an `id` should have a `fixedId` boolean also, but that
```
Comment (jakemac53) [inline dart/tool/dart/generate_dart.dart]: :tada:

**Pull request merged**